platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/http/OkHttpRequest.java

```diff
@@ -0,0 +1,246 @@
+package com.mapbox.mapboxsdk.http;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.v4.util.LongSparseArray;
+import android.text.TextUtils;
+import android.util.Log;
+import com.mapbox.android.telemetry.TelemetryUtils;
+import com.mapbox.mapboxsdk.BuildConfig;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.constants.MapboxConstants;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Dispatcher;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import timber.log.Timber;
+
+import javax.net.ssl.SSLException;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.net.NoRouteToHostException;
+import java.net.ProtocolException;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+
+public class OkHttpRequest extends HttpRequest {
+
+  private static final LongSparseArray<Call> calls = new LongSparseArray<>();
+  private static final String userAgentString =  TelemetryUtils.toHumanReadableAscii(
+    String.format("%s %s (%s) Android/%s (%s)",
+      getApplicationIdentifier(),
+      BuildConfig.MAPBOX_VERSION_STRING,
+      BuildConfig.GIT_REVISION_SHORT,
+      Build.VERSION.SDK_INT,
+      Build.CPU_ABI)
+  );
+
+  private static OkHttpClient client = new OkHttpClient.Builder().dispatcher(getDispatcher()).build();
+  private static boolean logEnabled = true;
+  private static boolean logRequestUrl = false;
+
+  @Override
+  public void executeRequest(NativeHttpRequest httpRequest, long nativePtr, String resourceUrl,
+                             String etag, String modified) {
+    Call call = null;
+    OkHttpCallback callback = new OkHttpCallback(httpRequest);
+    try {
+      HttpUrl httpUrl = HttpUrl.parse(resourceUrl);
+      if (httpUrl == null) {
+        log(Log.ERROR, String.format("[HTTP] Unable to parse resourceUrl %s", resourceUrl));
+      }
+
+      final String host = httpUrl.host().toLowerCase(MapboxConstants.MAPBOX_LOCALE);
```

Comment:
Logic like computing the host is not specific to OkHttp so I think it should be available in a delegate.